test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt

```diff
@@ -180,7 +189,77 @@ private suspend fun execute(
 
                                 if (indexOfCall != null) {
                                     val serviceCall = serviceCalls.removeAt(indexOfCall)
-                                    serviceCall.response
+                                    if (serviceCall.incrementalResponse != null && serviceCall.response != null) {
+                                        fail(
+                                            """Cannot have both an incremental and non-incremental response 
+                                        |   fixture : '${fixture.name}' 
+                                        |   service : '${serviceName}' 
+                                        |   service : '${serviceName}' 
+                                        |   query : '${actualQuery}' 
+                                        |   variables : '${actualVariables}' 
+                                        |   operation : '${actualOperationName}' 
+                                        """.trimMargin()
+                                        )
+                                    }
+                                    else if (serviceCall.incrementalResponse != null) {
+                                        fun transformData(executionResult: JsonMap): DelayedIncrementalPartialResult{
```

Comment:
I think this function needs a more specific name: `transformDelayedIncrementalPartialResult` or `transformDelayed`...
I'd also move its definition out of the `getServiceExecution` function for better readability.
The other transformation steps could also be extracted into their own functions: `transformExecutionResult`, `transformIncrementalExecutionResult`, etc.